src/Coffee/order/CoffeeOrderBoard.java

```diff
@@ -0,0 +1,52 @@
+package Coffee.order;
+
+import java.util.LinkedList;
+
+public class CoffeeOrderBoard {
+
+    private int lastOrderNumber;
+
+    private final LinkedList<Order> ordersInProgress = new LinkedList<>();
+
+
+
+    public void add(String nameClient) {
+        ordersInProgress.addLast(new Order(++lastOrderNumber, nameClient));
+    }
+
+    public Order deliver() {
+        return ordersInProgress.isEmpty() ? null : ordersInProgress.removeFirst();
+    }
+
+    public Order deliver(int orderNumber) {
+
+        if (ordersInProgress.isEmpty()) return null;
+        Order orderToDeliver = null;
+
+        int left = 0;
+        int right = ordersInProgress.size() - 1;
+        while (left <= right) {
```

Comment:
why just no to use for loop, it will be simpler and without all these pointers like left right